repair/repair.cc

```diff
@@ -397,30 +399,43 @@ static future<bool> flush_hints(repair_service& rs, repair_uniq_id id, replica::
         auto hints_timeout = std::chrono::seconds(300);
         auto batchlog_timeout = std::chrono::seconds(300);
         repair_flush_hints_batchlog_request req{id.uuid(), participants, hints_timeout, batchlog_timeout};
-
+        auto start_time = gc_clock::now();
+        std::vector<gc_clock::time_point> times;
         try {
-            co_await parallel_for_each(waiting_nodes, [&rs, uuid, &req, &participants] (gms::inet_address node) -> future<> {
+            co_await parallel_for_each(waiting_nodes, [this, uuid, start_time, &times, &req, &participants] (gms::inet_address node) -> future<> {
                 rlogger.info("repair[{}]: Sending repair_flush_hints_batchlog to node={}, participants={}, started",
                         uuid, node, participants);
                 try {
-                    auto& ms = rs.get_messaging();
+                    auto& ms = get_messaging();
                     auto resp = co_await ser::partition_checksum_rpc_verbs::send_repair_flush_hints_batchlog(&ms, netw::msg_addr(node), req);
-                    (void)resp; // nothing to do with response yet
+                    if (resp.flush_time == gc_clock::time_point()) {
+                        // This means the node does not support sending flush_time back. Use the time when the flush is requested for flush_time.
+                        rlogger.debug("repair[{}]: Got empty flush_time from node={}. Please upgrade the node={}.", uuid, node, node);
+                        times.push_back(start_time);
+                    } else {
+                        times.push_back(resp.flush_time);
+                    }
                 } catch (...) {
                     rlogger.warn("repair[{}]: Sending repair_flush_hints_batchlog to node={}, participants={}, failed: {}",
                             uuid, node, participants, std::current_exception());
                     throw;
                 }
             });
+            if (!times.empty()) {
+                auto it = std::min_element(times.begin(), times.end());
+                flush_time = *it;
+            }
```

Comment:
The participants will include the node itself at least, so it will not be empty. Also, it will be safer to set the flush time to zero by default.  If no real flush time is returned, we should not move the flush time to now.